### PR TITLE
Updated packages

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Updating packages, mainly:
- `opds-web-client`
- `typescript`
- `webpack`
- `@types` instead of `typings`

Also updated unit tests based on the `fetch-mock` mocking package.

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
Managed to get the production javascript build to go from 3.5MB down to 1.5MB.

**Comment by EdwinGuzman** [comment]
The app should build and run just as before so there should be no noticeable changes. No new features were added but some unit tests were updated.

File: src/server/__tests__/LibraryDataCache-test.ts [hunk 1]
```diff
@@ -40,42 +41,40 @@ describe("LibraryDataCache", () => {
...
           href: template
         }]
       };
-      let mockFetch = stub().returns(new Promise<any>((resolve) => {
-        resolve({ json: () => {
-          return registryCatalog;
-        }});
-      }));
-      fetch = mockFetch as any;
+      // Originally was { json: () => registryCatalog } for the return promise value
```
Comment (aslagle) [inline src/server/__tests__/LibraryDataCache-test.ts]: We don't need to keep track of this - git always has the history if we really want to know what it used to be.

Comment (EdwinGuzman) [reply src/server/__tests__/LibraryDataCache-test.ts]: Yeah, this was more for me while updating the tests to see the change. Can remove.

**Comment by heart-of-code** [comment]
> Looks good to me but I haven't tested it.
> 
> @heart-of-code, do you want to make sure you can run this before it's merged?

Looks good! Built and ran fine

**Pull request merged**